src/hapi-i18next.ts

```diff
@@ -82,23 +83,23 @@ export var register: HapiPluginRegister = function (server, options: any, next):
 			// Reads language if it was set from previous session or recently by client
 			temp = detectLanguageFromCookie(request);
 			language = trySetLanguage(temp);
-
-			if (!request.state[i18nextOptions.cookieName] || request.state[i18nextOptions.cookieName] !== language) {
```

Comment:
Just move this logic and add a check for `useCookie` into the callback for `i18n.setLng`